### PR TITLE
Gura - added test testOpenOrderPage

### PR DESCRIPTION
**Pull request by Gura-Andriy**

### CODE REVIEW

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 1]
```diff
@@ -41,6 +41,6 @@ public void testEditOrder() {
 
     @Test
     public void testOpenOrderPage() {
-        Assert.assertTrue(openOrderPage.isThisIsOrderPage());
+        Assert.assertTrue(openOrderPage.getPageTitle().equals("Оформлення замовлення"), "The page title shouldn't match the condition");
```
Comment (SofiiaPro) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: divide methods.
.getPageTitile()
.equals()

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 2]
```diff
@@ -41,6 +41,7 @@ public void testEditOrder() {
 
     @Test
     public void testOpenOrderPage() {
-        Assert.assertTrue(openOrderPage.isThisIsOrderPage());
+        Assert.assertTrue(!openOrderPage.getPageTitle().isEmpty(), "The title shouldn't found on page");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: shouldn't be empty

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 3]
```diff
@@ -41,6 +41,7 @@ public void testEditOrder() {
...
+        Assert.assertTrue(openOrderPage.getPageTitle().equals("Оформлення замовлення"), "The page title shouldn't match the condition");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: openOrderPage.getPageTitle() to var

Comment (toleksyn) [reply src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: should be

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 4]
```diff
@@ -9,7 +9,7 @@ public BasketPage editItemInBasket() {
         return new BasketPage();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:20]: resolve conflicts

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java (hunk 4 above)
Comment (Gura-Andriy) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:9]: I haven't conflicts O_O

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 5]
```diff
@@ -40,4 +41,23 @@ public int getOrderProductSum(int numberOfProduct) {
                 .text());
         return Integer.parseInt(ProductPrice.replace("₴", "").replace(" ", ""));
     }
+
+    public BasketPage deleteGoods(int indexOfGoods) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: you're deleting single product, rename
no 'of' in names

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 6]
```diff
@@ -40,4 +41,23 @@ public int getOrderProductSum(int numberOfProduct) {
...
+        $x(String.format("//button[@aria-controls='cartProductActions[%d]']", indexOfGoods)).click();
+        $x("//button[@class='button button--medium button--with-icon button--link context-menu-actions__button']").click();
+        return new BasketPage();
+    }
+
+    public BasketPage restoreDeletedGoods(int indexOfGoods) {
+        $$x("//app-buy-button[@class='toOrder']").get(indexOfGoods).click();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: inefficient, re-do

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java (hunk 5 above)
Comment (toleksyn) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: not good, product

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 7]
```diff
@@ -40,4 +41,23 @@ public int getOrderProductSum(int numberOfProduct) {
                 .text());
         return Integer.parseInt(ProductPrice.replace("₴", "").replace(" ", ""));
     }
+
+    public BasketPage deleteGoods(int indexOfGoods) {
+        $x(String.format("//button[@aria-controls='cartProductActions[%d]']", indexOfGoods)).click();
+        $x("//button[@class='button button--medium button--with-icon button--link context-menu-actions__button']").click();
+        return new BasketPage();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: return this;

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 8]
```diff
@@ -40,4 +41,23 @@ public int getOrderProductSum(int numberOfProduct) {
...
+        $x("//button[@class='button button--medium button--with-icon button--link context-menu-actions__button']").click();
+        return this;
+    }
+
+    public BasketPage restoreDeletedProduct(int productIndex) {
+        $x(String.format("(//button[@class='buy-button lite-tile__buy-button'])[%d]", productIndex)).click();
+        return this;
+    }
+
+    public String getDeletedFromBasketProductLink(int productIndex) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: rename

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 9]
```diff
@@ -40,4 +41,23 @@ public int getOrderProductSum(int numberOfProduct) {
...
+        return $x(String.format("(//a[@class='lite-tile__title'])[%d]", productIndex)).getText();
+    }
+
+    public String getProductInBasketLink(int productIndex) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/BasketTest.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+    //In progress
+    @Test
+    public void testRestoreDeletedOrder() {
+        openBasketPage.deleteProduct(0);
+        var deletedProductLink = openBasketPage.getDeletedFromBasketProductLink(0);
+        openBasketPage.restoreDeletedProduct(0);
+        var restoredProductLink = openBasketPage.getProductInBasketLink(0);
+        assertTrue(deletedProductLink == restoredProductLink, "Product shouldn't added to basket");
+
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/BasketTest.java]: remove

File: src/test/java/com/softserveinc/ita/rozetka_test/BasketTest.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    public void addProductToBasket() {
+        openBasketPage = new HomePage()
+                .openHomePage()
+                .openProductByNumber(1)
+                .addProductToBasket();
+    }
+
+    //In progress
+    @Test
+    public void testRestoreDeletedOrder() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/BasketTest.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 12]
```diff
@@ -21,4 +24,11 @@ public void addProductToBasketOpenOrder() {
                 .addProductToBasket()
                 .openOrderPage();
     }
+
+    @Test
+    public void testOpenOrderPage() {
+        var pageTitle = openOrderPage.getPageTitle();
+        assertTrue(!pageTitle.isEmpty(), "The title shouldn't be empty");
+        assertTrue(pageTitle.equals("Оформлення замовлення"), "The page title should match the condition");
+    }
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java:33]: re-design case

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java (hunk 8 above)
Comment (toleksyn) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: getRecentlyViewedProductsSection().getProduct(1).getLink()

**Pull request closed without merging**